eth/executionclient/execution_client.go

```diff
@@ -244,6 +244,15 @@ func (ec *ExecutionClient) Healthy(ctx context.Context) error {
 	return nil
 }
 
+func (ec *ExecutionClient) BlockTime(ctx context.Context, blockNumber uint64) (time.Time, error) {
```

Comment:
Just two notes for future-proofing the `ExecutionClient`:
1. Accept int64 instead of uint64 so it's possible to pass special numbers such as `rpc.FinalizedBlockNumber`
2. Consider renaming to `BlockTimeByNumber` or maybe just `BlockByNumber` and return the whole block.